src/extensionFunctions.ts

```diff
@@ -896,4 +905,35 @@ export function sendTextToNamedOutputChannel(message: string, channelName?: stri
 	if (!channelName || channelName === DIDACT_OUTPUT_CHANNEL) {
 		didactOutputChannel.show();
 	}
-}
\ No newline at end of file
+}
+
+export async function copyFileTextToClipboard(uri: vscode.Uri) : Promise<void> {
+	const testUri : vscode.Uri = uri;
+	const out : vscode.Uri | undefined = handleVSCodeDidactUriParsingForPath(testUri);
+	if (!out) {
+		const errmsg = `ERROR: No file found when parsing path ${uri}`;
+		sendTextToOutputChannel(errmsg);
+		vscode.window.showErrorMessage(errmsg);
+		throw new Error(errmsg);
+	} else {
+		let content : string | undefined = undefined;
+		if (out.scheme === 'file') {
+			try {
+				content = fs.readFileSync(out.fsPath, 'utf8');
+			} catch(error) {
+				showFileUnavailable(error);
+			}
+		} else if (out.scheme === 'http' || out.scheme === 'https'){
+			const urlToFetch = out.toString();
+			try {
+				const response = await fetch(urlToFetch);
+				content = await response.text();
+			} catch(error) {
+				showFileUnavailable(error);
+			}
+		}
```

Comment:
We are lacking an `else` here which actually is not a big issue but whenever someone tries with a protocol other than file or http(s) then there will be nothing in the logs that could help finding the problem from the users point of view. I would suggest to at least put out a message to the output channel or log to signal an unsupported scheme/protocol.

Reply:
Added the else case